src/Domain/ValueObjects/Wallet.cs

```diff
@@ -9,7 +9,7 @@ public record Wallet(decimal Balance)
 
     public bool HasBalance(decimal amount)
     {
-        if (amount == 0) throw new ArgumentException("amount can't be 0");
+        if (amount <= 0) throw new ArgumentException("amount must be positive and non zero");
```

Comment:
you can also pass the name of the argument which caused the exception

Reply:
which name?

Reply:
by argument name wdym?

Reply:
ArgumentException(nameof(amount), "msg")